intermittent unit test fail in test_unsorted

running pytest in the `refactor` branch, getting an intermittent failure in `tests/test_msg_processor.py::test_unsorted` at commit 21404f4fdd5bc599c7e9588a2a0dc92eeab72d78


Running on Paul's MBP M1

Steps to reproduce

```
git clone git@github.com:GlobalFishingWatch/gpsdio-segment.git
cd gpsdio-segment
docker compose build
docker compose run dev pytest
```

Usually I get

```
============================= test session starts ==============================
platform linux -- Python 3.10.3, pytest-7.1.1, pluggy-1.0.0
rootdir: /opt/project, configfile: setup.cfg, testpaths: tests
plugins: cov-3.0.0
collected 105 items                                                            

tests/test_bad_segment.py ..                                             [  1%]
tests/test_core.py ...                                                   [  4%]
tests/test_edge_cases.py ...............                                 [ 19%]
tests/test_expected.py .......ssssssssssssssssssssssssssssssssssssssssss [ 65%]
sssssssssssssssss                                                        [ 81%]
tests/test_identities.py ...                                             [ 84%]
tests/test_msg_processor.py .........                                    [ 93%]
tests/test_noise_segment.py .                                            [ 94%]
tests/test_rules.py ..                                                   [ 96%]
tests/test_state.py ....                                                 [100%]

======================== 46 passed, 59 skipped in 0.74s ========================

```

But sometimes I get 
```
============================= test session starts ==============================
platform linux -- Python 3.10.3, pytest-7.1.1, pluggy-1.0.0
rootdir: /opt/project, configfile: setup.cfg, testpaths: tests
plugins: cov-3.0.0
collected 105 items                                                            

tests/test_bad_segment.py ..                                             [  1%]
tests/test_core.py ...                                                   [  4%]
tests/test_edge_cases.py ...............                                 [ 19%]
tests/test_expected.py .......ssssssssssssssssssssssssssssssssssssssssss [ 65%]
sssssssssssssssss                                                        [ 81%]
tests/test_identities.py ...                                             [ 84%]
tests/test_msg_processor.py .....F...                                    [ 93%]
tests/test_noise_segment.py .                                            [ 94%]
tests/test_rules.py ..                                                   [ 96%]
tests/test_state.py ....                                                 [100%]

=================================== FAILURES ===================================
________________________________ test_unsorted _________________________________

    def test_unsorted():
        before = {
            "ssvid": 10,
            "msgid": 1,
            "timestamp": datetime.now(),
            "type": "UNKNOWN",
            "lat": 90,
            "lon": 90,
            "course": 0,
            "speed": 1,
        }
        after = {
            "ssvid": 10,
            "msgid": 2,
            "timestamp": datetime.now(),
            "type": "UNKNOWN",
            "lat": 90,
            "lon": 90,
            "course": 0,
            "speed": 1,
        }
        messages = [after, before]
        messages = [utcify(x) for x in messages]
        msg_processor = MsgProcessor(Matcher.very_slow, ssvid=10)
>       with pytest.raises(ValueError) as excinfo:
E       Failed: DID NOT RAISE <class 'ValueError'>

tests/test_msg_processor.py:259: Failed
```

--- a/tests/test_msg_processor.py
+++ b/tests/test_msg_processor.py
@@ -1,4 +1,5 @@
 from datetime import datetime
+from datetime import timedelta
 
 import pytest
 from support import utcify
@@ -246,7 +247,7 @@ def test_unsorted():
     after = {
         "ssvid": 10,
         "msgid": 2,
-        "timestamp": datetime.now(),
+        "timestamp": before["timestamp"] + timedelta(seconds=1),
         "type": "UNKNOWN",
         "lat": 90,
         "lon": 90,

